references/classification/train.py

```diff
@@ -306,6 +320,12 @@ def get_args_parser(add_help=True):
     parser.add_argument('--world-size', default=1, type=int,
                         help='number of distributed processes')
     parser.add_argument('--dist-url', default='env://', help='url used to set up distributed training')
+    parser.add_argument(
+        '--model-ema', action='store_true',
```

Comment:
Is there a reason why you introduce an extra param instead of setting the default of `--model-ema-decay` to 0.0? In that case your EMA will be as if you run the normal validation and thus you could omit it.

Reply:
I don't have a strong opinion on this, but I cant think of 2 reasons to have both --model-ema-decay and  the --model-ema flag as currently done:

- the flag allows for --model-ema-decay to have a good default of 0.99 which hopefully should work fine for most models (that might actually not be true in practice though, IDK). Without this, users would have to come up with a value themselves.
- it may not be obvious for inexperienced users that `ema_decay == 0 => no EMA`. The flag helps disambiguate whether EMA is used at all

Again, no strong opinion. Both of these could actually be solved with a few sentences in the parameter description.

Reply:
I guess @NicolasHug explained it well. Having `--model-ema` allows us to have a good default which users may choose to use. It also makes it simpler to use(not having to worry about choosing an ema_decay in your first run).